pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -337,13 +343,24 @@ func (c *Reconciler) createBuild(tr *v1alpha1.TaskRun, pvcName string) (*buildv1
 		return nil, err
 	}
 
+	build.Labels[knativeTaskRunLabelKey] = tr.Name
+	// Only propagate the pipeline and pipelinerun keys if they are there. If
+	// the TaskRun was created via PipelineRun these should be there.
+	if val, ok := tr.Labels[knativePipelineLabelKey]; ok {
+		build.Labels[knativePipelineLabelKey] = val
+	}
+	if val, ok := tr.Labels[knativePipelineRunLabelKey]; ok {
+		build.Labels[knativePipelineRunLabelKey] = val
```

Comment:
What is the value of adding the labels on the build object since the build itself will not take advantage of these labels for logging or something else?

Reply:
The only benefit at this point I can see is being able to query for builds related to a pipeline - `kubectl get builds --all-namespace -l pipeline.knative.dev/pipeline=somepipeline`

Reply:
ok, fair enough.